engine/src/main/java/org/terasology/rendering/dag/DebugCommands.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag;
+
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.console.commandSystem.annotations.Command;
+import org.terasology.logic.console.commandSystem.annotations.CommandParam;
+import org.terasology.logic.permission.PermissionManager;
+import org.terasology.rendering.dag.nodes.OutputToScreenNode;
+
+@RegisterSystem
+public class DebugCommands extends BaseComponentSystem {
```

Comment:
I don't see the need for this wrapper class. I stand by the WorldRendererImpl class as implementing ComponentSystem and hosting a single command that passes through information from the console to individual nodes. 

> Note that the functions right now are hard coded, but by making a map of <String, Node> to store <Command, ResponsibleNode> we can easily allow arbitrary commands to be registered by FBOs (not entirely sure how that would work with @command , but I'm sure I can work something out).

Meanwhile, what I'm suggesting would be even simpler: in WorldRendererImpl you just store a <nodeUri,node> map. A command arrives in the form of a string "myNodeUri myCommand myArgs", you fetch the node through the map and pass the string "myCommand myArgs" to the node. You don't even have to handle the specific command for the node, just pass it through, so that registration is much simpler.